scripts/process/as_svg.R

```diff
@@ -0,0 +1,266 @@
+
+#' Sub mustache keys into text
+#' 
+#' Use mustache templates to sub data.frame values into 
+#' strings. 
+#' 
+#' @param template a `template` per `whisker::whisker.render`: 
+#' "character with template text"
+#' @param data a data.frame with variables used for whisker rendering
+#' @details this function evaluates the render function row-wise. 
+#' Multiple keys can be combined in single template (see example). Per 
+#' whisker pkg, keys that don't have variables in `data` evaluate to empty 
+#' strings (no errors).
+#' @example 
+#' sub_attribute_text("hovertext('{{ID}}-{{dog}}' evt)", 
+#'    data.frame(ID = c('g','f'), dog = c('ralph','cindy')))
+#' 
+#' @return a character vector of length nrow(data)
+sub_attribute_text <- function(template, data){
+  sapply(seq_len(nrow(data)), FUN = function(j) {
+    whisker::whisker.render(template, data = data[j,])
+  }, USE.NAMES = FALSE)
+}
+
+#' sub text and/or define new svg attributes
+#' 
+#' uses `sub_attribute_text` to create attribute strings
+#' 
+#' @param dataframe data to use for evaluation
+#' @param attributes key value pairs that may or may not contain 
+#' mustache {{key}} syntax
+#' @return a data.frame with new columns for each name in `attributes`
+svg_sub_attributes <- function(dataframe, attributes = NULL){
+  
+  if (!is.null(attributes)){
+    # replace existing attributes w/ these user-specified ones
+    # evaluate "mustache" keys w/ geom_df
+    attr_names <- names(attributes)
+    
+    # if mustache keys exist, evaluate with the context of the dataframe variables:
+    attrs_subbed <- sapply(attr_names, FUN = function(x) {
+      sub_attribute_text(template = attributes[[x]], data = dataframe)
+    })
```

Comment:
Well this is nice to know - I always thought you had to pass in the other pieces of data to the function, so explicitly saying `function(x, attributes, dataframe)`. Good to know if just looks at the environment.